lib/service/service.go

```diff
@@ -5299,7 +5299,20 @@ func initSelfSignedHTTPSCert(cfg *servicecfg.Config) (err error) {
 	}
 	cfg.Log.Warningf("Generating self-signed key and cert to %v %v.", keyPath, certPath)
 
-	creds, err := cert.GenerateSelfSignedCert([]string{cfg.Hostname, "localhost"})
+	var hosts = []string{cfg.Hostname, "localhost"}
+
+	// add web public address hosts to self-signed cert
+	for _, addr := range cfg.Proxy.PublicAddrs {
+		proxyHost, _, err := net.SplitHostPort(addr.String())
+		if err != nil {
+			// log and skip error since this is a nice to have
+			cfg.Log.Errorf("Error parsing address %v, skipping adding to self-signed cert: %v", addr.String(), err)
```

Comment:
Nit:
Maybe we ought to name the configuration field more explicitly in the error log ? "address" without a qualifier can refer to a lot of things in Teleport.

```suggestion
			cfg.Log.Errorf("Error parsing proxy.public_address %v, skipping adding to self-signed cert: %v", addr.String(), err)
```